### ISSUE TITLE
StreamAddAsync gives error "ERR wrong number of arguments for XADD"

### ISSUE THREAD

**Issue by fmagn** [StackExchange/StackExchange.Redis#982]
I love it the new Redis Streams support! But I have a problem when passing in useApproximateMaxLength true.

This does not work (useApproximateMaxLength: true):
- redis.Database().StreamAddAsync("abc", "data", value, maxLength: 10, useApproximateMaxLength: **true**, CommandFlags.None);

When calling this way,  Rdis sends the error "ERR wrong number of arguments for XADD".

Thisis works (same call but with useApproximateMaxLength: false)
- redis.Database().StreamAddAsync("abc", "data", value, maxLength: 10, useApproximateMaxLength: **false**, CommandFlags.None);

**Comment by mgravell** [comment]
Fair enough - will investigate. At a cursory glance, it looks right
(GetStreamAddMessage appears to match https://redis.io/commands/xadd) - but
I don't doubt the report (rather,I'm just saying "hmmm, that's
odd/interesting")

**Comment by mgravell** [comment]
found, thanks (see above commit)

**Comment by mgravell** [comment]
release tied into #983

**Comment by fmagn** [comment]
Thanks for the references. I checked the code and it seems correct, but reading the "specification” for XADD, maybe a * is missing from the code after the length parameter (https://redis.io/commands/xadd)?

Specification:
XADD mystream MAXLEN ~ 1000 * abc ...

The code generates (I have not run the code in debug)
XADD mystream MAXLEN ~ 1000 abc …

Thanks,
Fredrik

**Comment by mgravell** [comment]
indeed; see the above commit - it isn't *missing*, but the `messageId` (which can be `*`) needs to *follow* the `MAXLEN` clause.

**Comment by fmagn** [comment]
Oh sorry, I didn't see your reply (replied by mail). Thank your very much!
